### PR TITLE
Shared: Add cancellable promise to Price component

### PR DESCRIPTION
**Pull request by tbergq**

### CODE REVIEW

File: packages/shared/src/CancellablePromise.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+export default (incomingPromise: Promise<*>) => {
+  let resolve;
+  let reject;
+
+  const promise = new Promise((originalResolve, originalReject) => {
+    resolve = originalResolve;
+    reject = originalReject;
+  });
+
+  incomingPromise.then(resolve).then(reject);
```
Comment (RobinCsl) [inline packages/shared/src/CancellablePromise.js]: I know this is just abstracting the original idea in the other file, but I don't fully understand the logic here. 
What is it useful for?

Comment (tbergq) [reply packages/shared/src/CancellablePromise.js]: basically, when you try to call setState on an unmounted component, you will get a yellow warning. It is also a memory leak. 

This is the suggested solution: 

https://reactjs.org/blog/2015/12/16/ismounted-antipattern.html

Comment (RobinCsl) [reply packages/shared/src/CancellablePromise.js]: If I look at the solution, it looks like it should be
```
incomingPromise.then(resolve, reject)
```

Comment (tbergq) [reply packages/shared/src/CancellablePromise.js]: That would not work. 

You can see #647 and we decieded to go for this solution:

https://blog.bloomca.me/2017/12/04/how-to-cancel-your-promise.html#pure-promises

**Pull request merged**